hyperspy/io_plugins/hdf5.py

```diff
@@ -425,9 +425,9 @@ def parse_structure(key, group, value, _type, compression):
             try:
                 group.attrs[key] = value
             except:
-                print("The hdf5 writer could not write the following "
-                      "information in the file")
-                print('%s : %s' % (key, value))
+                warnings.warn(
```

Comment:
It occurs when there is something wrong (i.e. we don't know how to save) in the metadata, yes. We don't suppress because it's not an exception as I understand it - we essentially check against our list of things that we can save, and if the item falls through, we just print it here. 

I've had it happen when I had to load a corrupted (I guess? never really got to the bottom of it) dm3 file, and the `original_metadata` was filled with things we could not save. 

It should not happen often, but when it happens I would rather still be able to save the main part (i.e. the data + axes).